src/server/acl/acl_family.cc

```diff
@@ -272,9 +280,68 @@ void AclFamily::Load(CmdArgList args, ConnectionContext* cntx) {
   cntx->SendOk();
 }
 
+void AclFamily::Log(CmdArgList args, ConnectionContext* cntx) {
+  if (args.size() > 1) {
+    (*cntx)->SendError(facade::OpStatus::OUT_OF_RANGE);
+  }
+
+  size_t max_output = 10;
+  if (args.size() == 1) {
+    auto option = facade::ToSV(args[0]);
+    if (absl::EqualsIgnoreCase(option, "RESET")) {
+      pool_->AwaitFiberOnAll([](auto index, auto* context) { ServerState::tlocal()->log.Reset(); });
+      (*cntx)->SendOk();
+      return;
+    }
+
+    if (!absl::SimpleAtoi(facade::ToSV(args[0]), &max_output)) {
+      (*cntx)->SendError("Invalid count");
+      return;
+    }
+  }
+
+  std::vector<AclLog::LogType> logs(pool_->size());
+  pool_->AwaitFiberOnAll(
+      [&logs](auto index, auto* context) { logs[index] = ServerState::tlocal()->log.GetLog(); });
+
+  size_t total_entries = 0;
+  for (auto& log : logs) {
+    total_entries += std::min(max_output, log.size());
+  }
+
+  (*cntx)->StartArray(total_entries);
+  for (auto& log : logs) {
```

Comment:
We should merge + sort by timestamp otherwise the first shard can shadow everything from the other shards.

Reply:
:gem: :100: